src/Microsoft.AspNetCore.Mvc.TagHelpers/AnchorTagHelper.cs

```diff
@@ -189,13 +200,12 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
                 if (Area != null)
                 {
-                    if (routeValues == null)
-                    {
-                        routeValues = new RouteValueDictionary();
-                    }
+                    SetRouteValue(ref routeValues, "area", Area);
+                }
 
-                    // Unconditionally replace any value from asp-route-area.
-                    routeValues["area"] = Area;
+                if (Page != null)
```

Comment:
This one wasn't super hard since it works with what we already had. `asp-page-sibling` is a bit less elegant since we'd have to post-process it here or call in to a RazorPages specific API.